dir.c

```diff
@@ -3448,6 +3448,83 @@ rb_dir_s_empty_p(VALUE obj, VALUE dirname)
  *  (<code>config.h</code> and <code>main.rb</code>), the parent
  *  directory (<code>..</code>), and the directory itself
  *  (<code>.</code>).
+ *
+ *  == What's Here
+ *
+ *  \Class \Dir provides methods that are useful for:
+ *
+ *  - {Creating}[#class-Dir-label-Creating]
+ *  - {Reading}[#class-Dir-label-Reading]
+ *  - {Setting}[#class-Dir-label-Setting]
+ *  - {Querying}[#class-Dir-label-Querying]
+ *  - {Iterating}[#class-Dir-label-Iterating]
+ *  - {Other}[#class-Dir-label-Other]
+ *
+ *  === Creating
+ *
+ *  - ::mkdir:: Creates and returns a new \Directory at the given path,
+ *              with optional permissions.
+ *  - ::new:: Creates and returns a new \Directory at the given path,
+ *            with optional encoding.
+ *
+ *  === Reading
+ *
+ *  - ::open:: Opens and returns a \Directory at the given path, with optional encoding;
```

Comment:
`Same as ::new, but if a block is given, yields the \Dir to the block, closing it upon block exit.`  This should also be moved to the Other section, since it doesn't actually read the directory, it just yields the Dir.